react/PhoneField.tsx

```diff
@@ -63,129 +63,134 @@ const renderCountryFlagWithCode = ({
-  )!
+    const updatePhone = (phone: string, rule: PhoneRuleDescriptor) => {
+      const updatedValue = rule.mask ? msk.fit(phone, rule.mask) : phone
 
-  const updatePhone = (phone: string, rule: PhoneRuleDescriptor) => {
-    const updatedValue = rule.mask ? msk.fit(phone, rule.mask) : phone
-
-    onChange({
-      value: `+${rule.countryCode}${unmaskPhone(updatedValue)}`,
-      isValid: !rule.mask || rule.mask.length === updatedValue.length,
-    })
-  }
+      onChange({
+        value: `+${rule.countryCode}${unmaskPhone(updatedValue)}`,
+        isValid: !rule.mask || rule.mask.length === updatedValue.length,
+      })
+    }
 
-  const handleChange: React.ChangeEventHandler<HTMLInputElement> = ({
-    target: { value: eventValue },
-  }) => {
-    updatePhone(eventValue, countryRule)
-  }
+    const handleChange: React.ChangeEventHandler<HTMLInputElement> = ({
+      target: { value: eventValue },
+    }) => {
+      updatePhone(eventValue, countryRule)
+    }
 
-  const handleCountryUpdate = (country: string) => {
-    const newRule = rules.find(({ countryISO }) => countryISO === country)!
+    const handleCountryUpdate = (country: string) => {
+      const newRule = rules.find(({ countryISO }) => countryISO === country)!
 
-    updatePhone(phoneData.phoneValue, newRule)
+      updatePhone(phoneData.phoneValue, newRule)
 
-    // schedule focus for the next tick since the listbox
-    // is already focused by default after this event handler,
-    // and we need to override that
-    setTimeout(() => void inputRef.current?.focus(), 0)
-  }
-
-  return (
-    <div className={styles.phoneField}>
-      <Input
-        {...props}
-        inputMode="numeric"
-        value={msk(phoneData.phoneValue, countryRule.mask ?? '')}
-        onChange={handleChange}
-        ref={(node: HTMLInputElement) => {
-          if (typeof ref === 'function') {
-            ref(node)
-          } else if (ref != null) {
-            // @ts-ignore: React TS types says this is read-only, but
-            // it its possible to mutate this value
-            ref.current = node
-          }
+      // schedule focus for the next tick since the listbox
+      // is already focused by default after this event handler,
+      // and we need to override that
+      setTimeout(() => void inputRef.current?.focus(), 0)
+    }
 
-          inputRef.current = node
-        }}
-        prefix={
-          <ListboxInput
-            className="h-100 flex-auto"
-            value={phoneData.selectedCountry}
-            onChange={handleCountryUpdate}
-          >
-            <ListboxButton
-              arrow={
-                <div className="c-action-primary flex items-center ml2">
-                  <ArrowDownIcon size={16} />
-                </div>
+    return (
+      <div className={styles.phoneField}>
+        <Input
+          {...props}
+          inputMode="numeric"
+          value={msk(phoneData.phoneValue, countryRule.mask ?? '')}
```

Comment:
The `msk` function returns an empty `string` if one of its param was empty (or false value).

Reply:
nice, thanks for the heads up!